lib/good_job/active_job_extensions/concurrency.rb

```diff
@@ -9,21 +9,21 @@ module Concurrency
       included do
         class_attribute :good_job_concurrency_config, instance_accessor: false, default: {}
 
-        before_enqueue do |job|
+        around_enqueue do |job, block|
           # Always allow jobs to be retried because the current job's execution will complete momentarily
-          next if CurrentExecution.active_job_id == job.job_id
+          block.call if CurrentExecution.active_job_id == job.job_id
```

Comment:
For each of these `block.calls` I think they need to instead be `next(block.call)` to return early and ensure that `block.call` is not called multiple times.

Reply:
You are absolutely right and that went completely over my head. That’s a really good catch!